sdk/src/wallet/core/mod.rs

```diff
@@ -46,15 +46,21 @@ use crate::{
 /// The stateful wallet used to interact with an IOTA network.
 #[derive(Debug)]
 pub struct Wallet<S: SecretManage = SecretManager> {
+    pub(crate) address: Arc<RwLock<Bech32Address>>,
```

Comment:
Do we even want these 3 to be mutable?

Reply:
Do we have an issue to remove the mutability at a later point then?

Reply:
create an issue: https://github.com/iotaledger/iota-sdk/issues/2031

Reply:
created an issue: https://github.com/iotaledger/iota-sdk/issues/2032